cpp/tests/utilities/test_utilities_impl.cuh

```diff
@@ -187,5 +195,61 @@ mg_graph_to_sg_graph(
   return std::make_tuple(std::move(graph), std::move(edge_weights), std::move(new_number_map));
 }
 
+template <typename vertex_t, bool multi_gpu>
+rmm::device_uvector<vertex_t> random_ext_vertex_ids(raft::handle_t const& handle,
+                                                    rmm::device_uvector<vertex_t> const& number_map,
+                                                    vertex_t count,
+                                                    uint64_t seed)
+{
+  rmm::device_uvector<vertex_t> vertex_ids(0, handle.get_stream());
+  std::vector<vertex_t> vertex_ends;
+
+  vertex_t number_of_vertices = static_cast<vertex_t>(number_map.size());
+  vertex_t base_index{0};
+
+  if constexpr (multi_gpu) {
+    auto vertex_sizes =
+      host_scalar_allgather(handle.get_comms(), number_of_vertices, handle.get_stream());
+
+    vertex_ends.resize(vertex_sizes.size());
+
+    std::inclusive_scan(vertex_sizes.begin(), vertex_sizes.end(), vertex_ends.begin());
+
+    number_of_vertices = vertex_ends.back();
+    int rank           = handle.get_comms().get_rank();
+    if (rank > 0) base_index = vertex_ends[rank - 1];
+  }
+
+  if (!multi_gpu || (handle.get_comms().get_rank() == 0)) {
+    vertex_ids.resize(count, handle.get_stream());
+
+    rmm::device_scalar<vertex_t> d_num_vertices(number_of_vertices, handle.get_stream());
+    raft::random::RngState rng_state(seed);
+
```

Comment:
We may need to consider when ```#ifdef NO_CUGRAPH_OPS``` is true.

And should this low level testing utility function be dependent on cugraph-ops? (and under the current cugraph-ops implementation, this will be served by 1 CUDA warp).

We may consider an alternative.

Say when count << num_vertices,

Generate count + alpha random indices and remove duplicates and take first count (generate additional random numbers if the resulting unique indices are fewer than count).

If count is in the same order of num_vertices, use thrust::shuffle to shuffle indices from 0 to num_vertices - 1 and take the first count indices.